lib/interceptAccess.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * interceptAccess.c
+ *
+ *  Created on: Oct 15, 2016
+ *      Author: ujay
+ */
+#define _GNU_SOURCE
+#include<dlfcn.h>
+#include<stdio.h>
+#include<sys/stat.h>
+#include<unistd.h>
+#include<errno.h>
+#include<string.h>
+#include<sys/types.h>
+#include<dirent.h>
+#include<fcntl.h>
+#include"../include/cwrapHeader.h"
+
+int (*real_faccessat)(int fd1,const char* pathname1,int mode,int flag)=NULL;
```

Comment:
Global variables actually don't need to be initialized to `NULL`: that happens automatically.